From b3ee7a761fb14385e6d7b44ef6f62d694ec5ba9e Mon Sep 17 00:00:00 2001
From: Marc Mutz <marc.mutz@qt.io>
Date: Tue, 30 Sep 2025 08:02:11 +0200
Subject: [PATCH 247/553] Port tst_QWinRegistryKey away from QPair

QPair is only a template alias for std::pair these days, and qMakePair
only a perfect forwarder for std::make_pair. There's no need to
continue using them, esp. since, when used, the source code will say
QPair while the debugger will say std::pair, and we'll want, at some
point, deprecate them.

So port from QPair / qMakePair to std::pair with CTAD.

As a drive-by, replace QStringView members with const wchar_t* ones
(both make_pair and CTAD decay the arguments), in preparation for a
follow-up change to the QWinRegistryKey API.

Task-number: QTBUG-115841
Pick-to: 6.8 6.5
Change-Id: I27690b309ce4f24e3bee52aa612cb8da4311df53
Reviewed-by: Matthias Rauter <matthias.rauter@qt.io>
(cherry picked from commit 7493986b0b9cdaf0ae3f0da12c982ada73bf2fd7)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 .../qwinregistrykey/tst_qwinregistrykey.cpp   | 19 +++++++++----------
 1 file changed, 9 insertions(+), 10 deletions(-)

diff --git a/tests/auto/corelib/kernel/qwinregistrykey/tst_qwinregistrykey.cpp b/tests/auto/corelib/kernel/qwinregistrykey/tst_qwinregistrykey.cpp
index 8125851c531..9a05de0dc92 100644
--- a/tests/auto/corelib/kernel/qwinregistrykey/tst_qwinregistrykey.cpp
+++ b/tests/auto/corelib/kernel/qwinregistrykey/tst_qwinregistrykey.cpp
@@ -3,7 +3,6 @@
 
 #include <QTest>
 #include <QObject>
-#include <QPair>
 #include <QScopeGuard>
 #include <private/qwinregistry_p.h>
 #include <qt_windows.h>
@@ -12,15 +11,15 @@ using namespace Qt::StringLiterals;
 
 static constexpr const wchar_t TEST_KEY[] = LR"(SOFTWARE\tst_qwinregistrykey)";
 
-static const QPair<QStringView, QString> TEST_STRING = qMakePair(u"string", u"string"_s);
-static const QPair<QStringView, QString> TEST_STRING_NULL = qMakePair(u"string_null", QString());
-static const QPair<QStringView, QStringList> TEST_STRINGLIST = qMakePair(u"stringlist", QStringList{ u"element1"_s, u"element2"_s, u"element3"_s });
-static const QPair<QStringView, QStringList> TEST_STRINGLIST_NULL = qMakePair(u"stringlist_null", QStringList());
-static const QPair<QStringView, quint32> TEST_DWORD = qMakePair(u"dword", 123);
-static const QPair<QStringView, quint64> TEST_QWORD = qMakePair(u"qword", 456);
-static const QPair<QStringView, QByteArray> TEST_BINARY = qMakePair(u"binary", "binary\0"_ba);
-static const QPair<QStringView, QVariant> TEST_NOT_EXIST = qMakePair(u"not_exist", QVariant());
-static const QPair<QStringView, QVariant> TEST_DEFAULT = qMakePair(u"", u"default"_s);
+const std::pair TEST_STRING{L"string", u"string"_s};
+const std::pair TEST_STRING_NULL{L"string_null", QString()};
+const std::pair TEST_STRINGLIST{L"stringlist", QStringList{u"element1"_s, u"element2"_s, u"element3"_s}};
+const std::pair TEST_STRINGLIST_NULL{L"stringlist_null", QStringList()};
+const std::pair TEST_DWORD{L"dword", 123};
+const std::pair TEST_QWORD{L"qword", 456};
+const std::pair TEST_BINARY{L"binary", "binary\0"_ba};
+const std::pair TEST_NOT_EXIST{L"not_exist", QVariant()};
+const std::pair TEST_DEFAULT{L"", u"default"_s};
 
 [[nodiscard]] static inline bool write(const HKEY key, const QStringView name, const QVariant &value)
 {
-- 
2.51.2

